client/client-multi/client-c/src/kaa/platform-impl/posix/file_utils.c

```diff
@@ -17,19 +17,32 @@
 #include <platform/file_utils.h>
 #include <stdint.h>
 #include <platform/stdio.h>
+#include <string.h>
+#include <platform/kaa_client_properties.h>
 #include "utilities/kaa_mem.h"
 #include "kaa_common.h"
 
-
 int posix_binary_file_read(const char *file_name, char **buffer, size_t *buffer_size, bool *needs_deallocation)
 {
-    KAA_RETURN_IF_NIL4(file_name, buffer, buffer_size, needs_deallocation, -1);
+    char *kaa_binary_file_path = KAA_CALLOC(1, strlen(kaa_get_working_directory()) + strlen(file_name) + 1);
```

Comment:
`strlen(kaa_get_working_directory()) + strlen(file_name) + 1` can be computed once to avoid copy-pasting and redundant calls.